### PR TITLE
Rename external representation of Statistical History

### LINKED ISSUE #420: Rename Statistical History to Vitals [History]
Because of aesthetics and because "Statistical History" is not exactly rolling off the tongue. Change external representation of this feature from "Statistical History" aka "StatHist" to "Vitals History" or "Vitals":

-  VM.stathist -> VM.vitals
-  -XX:+EnableStatHist -> -XX:+EnableVitals

To keep the change to a minimum and to keep review work low, rename only the externals. Leave class names etc as they were.

### PR DESCRIPTION
**Pull request by tstuefe**
Rename "Statistical History" to "Vitals" with a minimum of fuss:
1) rename command: VM.stathist -> VM.vitals
2) rename switches: "EnableStatHist" -> "EnableVitals" etc.

fixes #420

### CODE REVIEW

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

**Comment by tstuefe** [comment]
> Looks good.
> 

Thanks.

> When merging, please squash and change commit message to:
> "SapMachine #420: Rename external representation of Statistical History"
> 

okay.

> I think we should also backport this to 11 and 12 for the sake of consistency, shouldn't we?

Sure, thats the plan :)

File: src/hotspot/share/runtime/globals.hpp [hunk 1]
```diff
@@ -513,14 +513,14 @@ const size_t minimumSymbolTableSize = 1024;
           "Print additional debugging information from other modes")        \
                                                                             \
   /* SapMachine 2019-02-20 : stathist */                                    \
-  product(bool, EnableStatHist, true,                                       \
-          "Enable Statistics history")                                      \
+  product(bool, EnableVitals, true,                                         \
+          "Enable recording of vitals")                                     \
```
Comment (GoeLin) [inline src/hotspot/share/runtime/globals.hpp]: * I would use the same word as below: "Enable __sampling__ of vitals"
  (or is the sampling on even if this flag is off?)
* I would give a short explanation of what vitals are:
"Enable sampling of vitals: memory, heap and cpu utilization" (Please use your own text ... this is only intended as example).

File: src/hotspot/share/services/stathist.cpp [hunk 2]
```diff
@@ -1132,7 +1132,7 @@ void cleanup() {
 
 void print_report(outputStream* st, const print_info_t* pi) {
 
-  st->print("stathist:");
+  st->print("Vitals History:");
```
Comment (GoeLin) [inline src/hotspot/share/services/stathist.cpp]: I would omit the word "History".  As I understand, there is no printout of a single sample which would justify to distinguish "Vitals" and "Vitals History"?

File: src/hotspot/share/services/stathistDCmd.hpp [hunk 3]
```diff
@@ -41,10 +41,10 @@ class StatHistDCmd : public DCmdWithParser {
 public:
   StatHistDCmd(outputStream* output, bool heap);
   static const char* name() {
-    return "VM.stathist";
+    return "VM.vitals";
   }
   static const char* description() {
-    return "Provide statistics history.";
+    return "Print Vitals History.";
```
Comment (GoeLin) [inline src/hotspot/share/services/stathistDCmd.hpp]: omit "History" too.

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

File: src/hotspot/share/runtime/globals.hpp (hunk 1 above)
Comment (tstuefe) [reply src/hotspot/share/runtime/globals.hpp]: Thanks Goetz. All good proposals. Will post update.

**Comment by SapMachine** [comment]
Hello @tstuefe, this pull request fulfills all formal requirements.

**Pull request merged**